File: data-jpa-post-service/src/main/java/com/example/repository/CommentRepository.java
package com.example.repository;

import com.example.domain.Comment;
import com.example.domain.Post;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

@Repository
public interface CommentRepository extends JpaRepository<Comment, UUID> {

    List<Comment> findByPost(Post post);
}

File: data-r2dbc-kotlin-co/src/main/kotlin/com/example/CommentRepository.kt
package com.example

import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.r2dbc.annotation.R2dbcRepository
import io.micronaut.data.repository.jpa.kotlin.CoroutineJpaSpecificationExecutor
import io.micronaut.data.repository.kotlin.CoroutineCrudRepository
import kotlinx.coroutines.flow.Flow
import java.util.*

@R2dbcRepository(dialect = Dialect.POSTGRES)
interface CommentRepository : CoroutineCrudRepository<Comment, UUID>, CoroutineJpaSpecificationExecutor<Comment> {
    fun findByPost(post: Post): Flow<Comment>
}